File: Src/Frontend/TaskMaster.UI/src/app/core/auth/auth-dependency.factory.ts
import { AuthDependency } from './auth.dependency';
import { NoAuthDependency } from './no-auth/no-auth.dependency';
import { OidcAuthDependency } from './oidc/oidc-auth.dependency';
import { AuthConfig, AuthMode } from '../models';

export function createAuthDependency(authConfig: AuthConfig): AuthDependency {
  switch (authConfig.mode) {
    case AuthMode.None:
      return new NoAuthDependency();
    case AuthMode.Oidc:
      return new OidcAuthDependency();
    default:
      throw new Error(`Unsupported auth mode: ${authConfig.mode}`);
  }
}

File: Src/Backend/TaskMaster.API/Auth/AuthDependencyFactory.cs
using TaskMaster.API.Auth.NoAuth;
using TaskMaster.API.Auth.Oidc;
using TaskMaster.API.Configs;

namespace TaskMaster.API.Auth;

public static class AuthDependencyFactory
{
    public static IAuthDependency Create(AuthMode mode) => mode switch
    {
        AuthMode.None => new NoAuthDependency(),
        AuthMode.Oidc => new OidcDependency(),
        _ => throw new InvalidOperationException($"Unsupported auth mode: {mode}.")
    };
}
